### PR TITLE
GEODE-5607: Replace Jmock with Mockito

### PR DESCRIPTION
**Pull request by jujoramos**
GEODE-5607: Replace Jmock with Mockito

- Removed dependency to `org.jmock`.
- Migrated several test classes to use `Mockito` instead `Jmock`.
- Changed same classes to use `assertj` instead of `junit.Assert`.
- Changed same classes to use `Awaitility` instead of `WaitCriterion`.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [X] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-web/src/test/java/org/apache/geode/management/internal/web/controllers/support/LoginHandlerInterceptorJUnitTest.java [hunk 1]
```diff
@@ -89,52 +79,66 @@ public T nextElement() {
...
-    assertNotNull(envSet);
-    assertNotSame(envBefore, envSet);
-    assertEquals(1, envSet.size());
-    assertTrue(envSet.containsKey("variable"));
-    assertEquals("two", envSet.get("variable"));
+    LoginHandlerInterceptor handlerInterceptor = new LoginHandlerInterceptor(securityService);
+    assertThat(handlerInterceptor.preHandle(mockHttpRequest, null, null)).isTrue();
+    Map<String, String> environmentAfterPreHandle = LoginHandlerInterceptor.getEnvironment();
+    assertThat(environmentAfterPreHandle).isNotNull();
+    assertThat(1).isEqualTo(environmentAfterPreHandle.size());
```
Comment (pdxrunner) [inline geode-web/src/test/java/org/apache/geode/management/internal/web/controllers/support/LoginHandlerInterceptorJUnitTest.java]: Looking at the rest of this test class, all of your `assertThat(<actual>).is*(expected)` have the `actual` and `expected` values swapped. Please review the rest of the tests were refactored to use assertj for similar swapping of values.

**Comment by jujoramos** [comment]
Hello @pdxrunner,
Thanks for reviewing the PR. Good catch, I'll make the changes and add another commit 👍 .
Cheers.

File: geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsJUnitTest.java [hunk 2]
```diff
@@ -103,339 +100,111 @@ private DiskStoreDetails createDiskStoreDetails(final String memberId,
...
-    final DiskStoreDetails diskStoreDetails2 =
-        createDiskStoreDetails("memberOne", "gatewayDiskStore");
     final DiskStoreDetails diskStoreDetails3 = createDiskStoreDetails("memberTwo", "pdxDiskStore");
     final DiskStoreDetails diskStoreDetails4 =
         createDiskStoreDetails("memberTwo", "regionDiskStore");
-
+    final DiskStoreDetails diskStoreDetails2 =
+        createDiskStoreDetails("memberOne", "gatewayDiskStore");
+    final DiskStoreDetails diskStoreDetails1 =
+        createDiskStoreDetails("memberOne", "cacheServerDiskStore");
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsJUnitTest.java]: Not a problem per se, but this is an odd reordering of the `DiskStoreDetails` declarations.

**Comment by jujoramos** [comment]
Hey @pdxrunner,
Thanks for reviewing this again, I'll make the changes shortly 👍 .
Cheers.

**Pull request merged**